src/components/Movies/MovieForm.tsx

```diff
@@ -0,0 +1,105 @@
+import React, {FC, useState} from 'react'
+
+import Modal from '../UI/Modal'
+import Button from '../UI/Button'
+
+import { Movie } from './MovieCard'
+
+import styles from './MovieForm.scss'
+import GenresSelect from '../UI/GenresSelect'
+
+interface MovieFormProps extends Movie {
+  heading: string
+  onClose: () => void
+}
+
+const MovieForm: FC<MovieFormProps> = ({
+  heading,
+  title= '',
+  rating= null,
+  runtime= null,
+  releaseDate= '',
+  overview= '',
+  genres= [],
+  onClose,
+}) => {
+  const [inputTitle, setInputTitle] = useState(title)
```

Comment:
Sorry, I'm not totally clear about that refactoring. Are you talking about props?

Reply:
No. Instead of having X `useState` usages you could just have only 1 which contains `values` where each key is the name of the input.

E.g. `setValues(prevValues => { ...prevValues, [e.target.name]: e.target.value })`.
This would result in something like `values.rating`, `values.releaseDate` and so on.

Reply:
Done